src/rp/index.ts

```diff
@@ -121,6 +127,85 @@ export class RelyingPartySolution {
     );
   }
 
+  /**
+   * Prepare the Verified Presentation token for a received request object in the context of an authorization request flow.
+   * The presentation is prepared by disclosing data from provided credentials, according to requested claims
+   * Each Verified Credential come along with the claims the user accepts to disclose from it.
+   *
+   * The returned token is unsigned (sign should be apply by the caller).
+   *
+   * @todo accept more than a Verified Credential
+   *
+   * @param requestObj The incoming request object, which the requirements for the requested authorization
+   * @param presentation The Verified Credential containing user data along with the list of claims to be disclosed.
+   * @returns The unsigned Verified Presentation token
+   * @throws {ClaimsNotFoundBetweenDislosures} If the Verified Credential does not contain one or more requested claims.
+   *
+   */
+  prepareVpToken(
+    requestObj: RequestObject,
+    [vc, claims]: Presentation // TODO: [SIW-353] support multiple presentations
+  ): string {
+    // this throws if vc cannot satisfy all the requested claims
+    const vp = disclose(vc, claims);
+
+    // check all requeste claims of the requestedObj are satisfied
+
+    const vp_token = new SignJWT({ vp })
+      .setAudience(requestObj.payload.response_uri)
+      .setExpirationTime("1h")
+      .setProtectedHeader({
+        typ: "JWT",
+        alg: "ES256",
+      })
+      .toSign();
+
+    return vp_token;
+  }
+
+  /**
+   * Compose and send an Authorization Response in the context of an authorization request flow.
+   *
+   * @todo MUST encrypt response payload
+   * @todo MUST add presentation_submission
+   *
+   * @param requestObj The incoming request object, which the requirements for the requested authorization
+   * @param vp_token The signed Verified Presentation token with data to send.
+   * @returns The response from the RP
+   * @throws IoWalletError if the submission fails.
+   *
+   */
+  async sendAuthorizationResponse(
+    requestObj: RequestObject,
+    vp_token: string
+  ): Promise<string> {
+    // the request is an unsigned jws without iss, aud, exp
+    // https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#name-signed-and-encrypted-respon
+    // TODO: [SIW-351] MUST be encrypted
+    const authzResponse = new SignJWT({
+      state: requestObj.payload.state,
+      // TODO: [SIW-352] MUST add presentation_submission
+      // presentation_submission:
+      vp_token,
+    }).toSign();
```

Comment:
since it will be a JWE, so the payload is a pure json...
```suggestion
    const authzResponsePayload = {
      state: requestObj.payload.state,
      // TODO: [SIW-352] MUST add presentation_submission
      // presentation_submission:
      vp_token,
    };
```